src/test/regress/expected/multi_view.out

```diff
@@ -883,3 +883,404 @@ DROP VIEW recent_selected_users;
+INSERT INTO small VALUES(14, 14);
+INSERT INTO large_partitioned VALUES(14, 14);
+-- test on a router executable update statement
+UPDATE large_partitioned SET id=28 FROM small_view WHERE small_view.id=large_partitioned.id and small_view.tenant_id=14 and large_partitioned.tenant_id=14;
+SELECT * FROM large_partitioned ORDER BY 1, 2;
+ id | tenant_id 
+----+-----------
+  6 |         5
+ 26 |        32
+ 27 |         2
+ 27 |         3
+ 27 |         4
+ 28 |        14
+ 29 |        15
+ 60 |        51
+(8 rows)
+
+-- we should still have identical rows for next test statements, then insert a new row to large_partitioned table
+INSERT INTO large_partitioned VALUES(14, 14);
+-- delete statement on large
+DELETE FROM large_partitioned WHERE tenant_id in (SELECT tenant_id FROM small_view);
+SELECT * FROM large_partitioned ORDER BY 1, 2;
+ id | tenant_id 
+----+-----------
+  6 |         5
+ 26 |        32
+ 29 |        15
+ 60 |        51
+(4 rows)
+
+-- we should still have identical rows for next test statement, then insert a new row to large table
+INSERT INTO large_partitioned VALUES(14, 14);
+-- delete statement with CTE
+WITH all_small_view_tenant_ids AS (SELECT tenant_id FROM small_view)
+DELETE FROM large_partitioned WHERE tenant_id in (SELECT * FROM all_small_view_tenant_ids);
+SELECT * FROM large_partitioned ORDER BY 1, 2;
+ id | tenant_id 
+----+-----------
+  6 |         5
+ 26 |        32
+ 29 |        15
+ 60 |        51
+(4 rows)
+
+DROP TABLE large_partitioned;
+DROP TABLE small CASCADE;
+NOTICE:  drop cascades to view small_view
+-- perform similar tests with a little bit complicated view
+-- create two tables and a view
+CREATE TABLE large (id int, tenant_id int);
+-- constraint id to be unique for "insert into on conflict" test
+CREATE TABLE small (id int, tenant_id int, unique(tenant_id));
+SELECT create_distributed_table('large','tenant_id');
+ create_distributed_table 
+--------------------------
+ 
+(1 row)
+
+SELECT create_distributed_table('small','tenant_id');
+ create_distributed_table 
+--------------------------
+ 
+(1 row)
+
+CREATE VIEW small_view AS SELECT id, tenant_id FROM (SELECT *, id*2 FROM small WHERE id < 100 ORDER BY 1,2 LIMIT 5) as foo;
+\copy small FROM STDIN DELIMITER ','
+\copy large FROM STDIN DELIMITER ','
+-- using views in modify statements' FROM / WHERE clauses is still valid 
+UPDATE large SET id=20 FROM small_view WHERE small_view.id=large.id;
+SELECT * FROM large order by 1, 2;
+ id | tenant_id 
+----+-----------
+  2 |         3
+  5 |         4
+ 20 |         2
+ 20 |         5
+(4 rows)
+
+-- we should still have identical rows for next test statements, then insert new rows to both tables
+INSERT INTO large VALUES(14, 14);
+INSERT INTO small VALUES(14, 14);
+-- using views in subqueries within modify statements is still valid 
+UPDATE large SET id=23 FROM (SELECT *, id*2 from small_view) as small_view WHERE small_view.id=large.id;
```

Comment:
This subquery is too simple. Postgres has a logic to pull this subquery into a join (e.g., get rid of the subquery, see `pull_up_subqueries` function in postgres source). To prevent postgres to pull up the subquery, can you add `ORDER BY ... LIMIT ..` to the subquery itself?

Reply:
Just read the leading comment of `pull_up_subqueries`: `If the subquery has no special features like grouping/aggregation then we can merge it into the parent's jointree.`

I am adding these extra clauses now